e2e/relayer/relayer.go

```diff
@@ -51,6 +57,87 @@ func New(t *testing.T, cfg Config, logger *zap.Logger, dockerClient *dockerclien
 	}
 }
 
+// ApplyPacketFilter applies a packet filter to the hermes config file, which specifies a complete set of channels
+// to watch for packets.
+func ApplyPacketFilter(ctx context.Context, t *testing.T, r ibc.Relayer, chainID string, channels []ibc.ChannelOutput) error {
+	t.Helper()
+
+	h, ok := r.(*hermes.Relayer)
+	if !ok {
+		t.Logf("relayer %T does not support packet filtering, or it has not been implemented yet.", r)
+		return nil
+	}
+
+	return modifyHermesConfigFile(ctx, h, func(config map[string]interface{}) error {
+		chains, ok := config["chains"].([]map[string]interface{})
+		if !ok {
+			return fmt.Errorf("failed to get chains from hermes config")
+		}
+		var chain map[string]interface{}
+		for _, c := range chains {
+			if c["id"] == chainID {
+				chain = c
+				break
+			}
+		}
+
+		if chain == nil {
+			return fmt.Errorf("failed to find chain with id %s", chainID)
+		}
+
+		var channelIDs [][]string
+		for _, c := range channels {
+			channelIDs = append(channelIDs, []string{c.PortID, c.ChannelID})
+		}
+
+		// [chains.packet_filter]
+		//	# policy = 'allow'
+		//	# list = [
+		//	#   ['ica*', '*'],
+		//	#   ['transfer', 'channel-0'],
+		//	# ]
+
+		// TODO(chatton): explicitly enable watching of ICA channels
+		// this will ensure the ICA tests pass, but this will need to be modified to make sure
+		// ICA tests will succeed in parallel.
+		channelIDs = append(channelIDs, []string{"ica*", "*"})
+
+		// we explicitly override the full list, this allows this function to provide a complete set of channels to watch.
+		chain["packet_filter"] = map[string]interface{}{
+			"policy": "allow",
+			"list":   channelIDs,
+		}
+
+		return nil
+	})
+}
+
+// modifyHermesConfigFile reads the hermes config file, applies a modification function and returns an error if any.
+func modifyHermesConfigFile(ctx context.Context, h *hermes.Relayer, modificationFn func(map[string]interface{}) error) error {
```

Comment:
Not sure if it would be any better or not, but interchaintest has some utility functions for modifying Toml files:
https://github.com/strangelove-ventures/interchaintest/blob/main/testutil/toml.go

Reply:
good call, I think I reinvented the wheel here!

Reply:
https://github.com/strangelove-ventures/interchaintest/issues/1154